### ISSUE TITLE
Pass expected parameters to prepare_commit_msg

### ISSUE THREAD

**Issue by samdeane** [ForkIssues/Tracker#1789]
When the git command line runs `prepare-commit-hook`, it is passed up to three parameters (see https://git-scm.com/docs/githooks#_prepare_commit_msg).

It looks like that's currently not happening when Fork runs it. It would be helpful if it did happen, so that existing hook scripts worked as expected.

Extra information, such as the existing "subject" and "description" field values, could also be passed as environment variables, eg: FORK_SUBJECT, and FORK_DESCRIPTION. 

This would be helpful as it would allow the script to know that it was running from Fork.app, and modify its behaviour accordingly (if required).

**Comment by svoop** [comment]
Indeed, the way Fork uses the `prepare-commit-hook` is currently incompatible with CLI Git:

* Fork expects the script to echo the commit message template.
* Git passes the default commit message as a file with the first argument and expects the script to edit the contents of this file.

The second argument contains the source of commit, either `message`, `template`, `merge`, `squash` or `commit` (which requires a third argument). IMO, only the first and the last of these are relevant for Fork.

Furtunately, it's not very difficult to make things compatible. Fork would have to...

* [ ] Set the first argument to an empty, temporary file. The `prepare-commit-hook` then edits this file and exits, at which point Fork reads the temporary file and prefills the input with this content.
* [ ] If the amend checkbox is NOT selected: Set the second argument to `message` and not set any third argument.
* [ ] If the amend checkbox IS selected: Set the second argument to `commit` and the third argument to the SHA of the commit to be amended.

/cc @DanPristupov

**Comment by Samhayne** [comment]
This is really painful as a prepare_commit_msg hook that was provided by my company doesn't work.
(The hook is supposed to automatically modify the commit subject, so it starts with the ticket number extracted from the branch name, so you can never forget that)

Have to switch to a different client now due to that. :/


edit: Odd. A few days later I use Fork again briefly, accidently committed without adding the prefix manually - and all of a sudden hook seems to work. ? (Version 2.8.2.0)

Tested it with test commits before and the hook wouldn't work. Now it does all of a sudden.
Yet, running it manually via "run prepare commit hook" from the context menu next to the commit subject still produces an error message box due to not receiving any of the $1 $2 $3 parameters.
